### PR TITLE
Add interactive mode

### PR DESCRIPTION
**Pull request by sava-cska**
Now interactive mode doesn't work on generation tests for project or folder, but for file all are fine. For file `linked-list.cpp` generation time equals 52s for 1 parallel process, 36s - for 5 and 20s - for 10. For other files it's impossible to see difference because number of functions in files are small

### CODE REVIEW

**Comment by korifey** [comment]
- Please show the difference with and without interactive mode for file with 100 functions
- why it doesn't work for folder?

**Comment by sava-cska** [comment]
Generation on file `big-linked-list.c` with 100 functions spends 981 seconds without interactive mode, 948 seconds with 1 process, 485 with 5 process and 262 seconds with 10 process

**Comment by korifey** [comment]
please make a table in markdown with column:
1. Number of functions in file (1, 5, 10, 50, 100, 500, 1000)
2. Execution time without interactive in sec
3. Execution time with interactive in sec

File: server/test/framework/External_Call_Test.cpp [hunk 1]
```diff
@@ -0,0 +1,4 @@
+//
+// Created by gWX1077059 on 20.12.2021.
```
Comment (ladisgin) [inline server/test/framework/External_Call_Test.cpp:2]: delete this file

File: server/src/Tests.cpp [hunk 2]
```diff
@@ -850,7 +850,9 @@ KTestObjectParser::parseTestCaseParams(const UTBotKTest &ktest,
         processGlobalParamPostValue(testCaseDescription, globalParam, rawKleeParams);
     }
 
-    processSymbolicStdin(testCaseDescription, rawKleeParams);
+    //We should turn off symbolic stdin, because it doesn't work correct with interactive mode
```
Comment (ladisgin) [inline server/src/Tests.cpp:853]: add TODO

**Review by ladisgin — CHANGES_REQUESTED** [review]

File: server/test/framework/External_Call_Test.cpp (hunk 1 above)
Comment (sava-cska) [reply server/test/framework/External_Call_Test.cpp:2]: Ok

File: server/src/Tests.cpp (hunk 2 above)
Comment (sava-cska) [reply server/src/Tests.cpp:853]: Ok

File: server/src/tasks/BaseForkTask.cpp [hunk 3]
```diff
@@ -174,8 +175,13 @@ int BaseForkTask::waitForFinishedOrCancelled() {
...
                 LOG_S(DEBUG) << "Sending signal to " << processName << ": " << signal;
-                int killStatus = kill(pid, signal);
-                if (killStatus == -1) {
+
+                pid_t pgid = getpgid(pid);
+                std::string pkillParent = "-g " + std::to_string(pgid);
+                std::string pkillSignal = "-" + std::to_string(signal);
+                std::string pkillCommand = "pkill " + pkillParent + " " + pkillSignal;
+                int killChildStatus = std::system(pkillCommand.c_str());
+                if (killChildStatus == -1 || WEXITSTATUS(killChildStatus) > 1) {
```
Comment (ladisgin) [inline server/src/tasks/BaseForkTask.cpp:184]: also add check WIFEXITED before WEXITSTATUS

**Review by ladisgin — CHANGES_REQUESTED** [review]

File: server/src/tasks/BaseForkTask.cpp [hunk 4]
```diff
@@ -45,9 +45,10 @@ ExecUtils::ExecutionResult BaseForkTask::run() {
             throw BaseException(message);
         }
         case 0: {
+            int pgidStatus = setpgid(pid, pid);
             grpc_postfork_child();
```
[the remark this answers is not in the document]
Comment (sava-cska) [reply server/src/tasks/BaseForkTask.cpp:52]: Ok

File: server/src/tasks/BaseForkTask.cpp [hunk 5]
```diff
@@ -171,8 +172,13 @@ int BaseForkTask::waitForFinishedOrCancelled() {
                 int signal = shutDownSignals[signalId];
                 throwIfNoSuchProcess();
                 LOG_S(DEBUG) << "Sending signal to " << processName << ": " << signal;
-                int killStatus = kill(pid, signal);
-                if (killStatus == -1) {
+
+                pid_t pgid = getpgid(pid);
+                std::string pkillParent = "-g " + std::to_string(pgid);
+                std::string pkillSignal = "-" + std::to_string(signal);
+                std::string pkillCommand = "pkill " + pkillParent + " " + pkillSignal;
```
[the remark this answers is not in the document]
Comment (sava-cska) [reply server/src/tasks/BaseForkTask.cpp:182]: Ok

File: server/src/tasks/BaseForkTask.cpp [hunk 6]
```diff
@@ -45,9 +45,10 @@ ExecUtils::ExecutionResult BaseForkTask::run() {
             throw BaseException(message);
         }
         case 0: {
+            int pgidStatus = setpgid(pid, pid);
             grpc_postfork_child();
             // This is child process
-            if (!redirectOutput()) {
+            if (!redirectOutput() || pgidStatus != 0) {
```
[the remark this answers is not in the document]
Comment (sava-cska) [reply server/src/tasks/BaseForkTask.cpp:54]: Ok

File: server/src/tasks/BaseForkTask.cpp (hunk 3 above)
Comment (sava-cska) [reply server/src/tasks/BaseForkTask.cpp:184]: Ok

File: server/src/Tests.cpp [hunk 7]
```diff
@@ -852,7 +852,9 @@ KTestObjectParser::parseTestCaseParams(const UTBotKTest &ktest,
         processGlobalParamPostValue(testCaseDescription, globalParam, rawKleeParams);
     }
 
-    processSymbolicStdin(testCaseDescription, rawKleeParams);
+    //TODO: We should turn off symbolic stdin, because it doesn't work correct with interactive mode
```
Comment (ladisgin) [inline server/src/Tests.cpp:855]: Change top "we need modify symbolic stdin for work with interactive mode"

File: server/src/tasks/BaseForkTask.cpp [hunk 8]
```diff
@@ -49,10 +50,16 @@ ExecUtils::ExecutionResult BaseForkTask::run() {
...
             grpc_postfork_child();
+            int pgidStatus = setpgid(pid, pid);
             // This is child process
             if (!redirectOutput()) {
                 exit(LOG_FAIL_CODE);
             }
+            if (pgidStatus != 0) {
+                std::string message = StringUtils::stringFormat("Failed setpgid(pid = %d, pgid = %d)");
+                LOG_S(ERROR) << message << LogUtils::errnoMessage();
+                exit(pgidStatus);
```
Comment (ladisgin) [inline server/src/tasks/BaseForkTask.cpp:61]: use BaseForTask fail code

**Review by ladisgin — CHANGES_REQUESTED** [review]

File: server/src/tasks/BaseForkTask.cpp (hunk 8 above)
Comment (sava-cska) [reply server/src/tasks/BaseForkTask.cpp]: Done

**Review by ladisgin — APPROVED** [review]

**Pull request merged**